### PR TITLE
Add new aggregation (min, max, sum, value_count) description to docs

### LINKED ISSUE #2737: Aggregations docs are missing some important details
**Describe the bug**
Aggregations docs are missing newly added `value_count`, `min`, `max`, and `sum` aggregations (see quickwit-oss/tantivy#1794). They don't mention support for `datetime` type for some aggregations and doesn't list supported types for the `terms` aggregation.

### PR DESCRIPTION
**Pull request by imotov**
### Description

Adds descriptions of newly added aggregations and clarifies some limitations and supported field types.

Fixes #2737

### How was this PR tested?

By manually running the queries and checking the tantivy code.

### CODE REVIEW

**Comment by fmassot** [comment]
thanks @imotov. @PSeitz Can you check that?

File: docs/reference/aggregation.md [hunk 1]
```diff
@@ -512,10 +513,109 @@ Supported field types are u64, i64, and f64.
 }
 ```
 
+### Count
+
+A single-value metric aggregation that computesthat counts the number of values that are extracted from the aggregated documents.
```
Comment (PSeitz) [inline docs/reference/aggregation.md:518]:
```suggestion
A single-value metric aggregation that counts the number of values that are extracted from the aggregated documents.
```

File: docs/reference/aggregation.md [hunk 2]
```diff
@@ -115,7 +114,9 @@ Some define a single bucket, some define a fixed number of multiple buckets, and
 
 Example request, histogram with stats in each bucket:
 
-#### Datetime Histogram Example
+#### Aggregating on datetime fields
+
+The `datetime` fields are handled the same way as other numric fields. However, all values in the requests such is intervals, offsets, bounds, and range boundaries need to be expressed in microseconds.
```
Comment (PSeitz) [inline docs/reference/aggregation.md:119]:
```suggestion
Fields of type `datetime` are handled the same way as any numeric field. However, all values in the requests such as intervals, offsets, bounds, and range boundaries need to be expressed in microseconds.
```

**Review by PSeitz — APPROVED** [review]

**Comment by PSeitz** [comment]
Thanks! fixed some typos, LGTM

**Pull request merged**